File: apple/Sources/ReactNativeWatchOS/Modifiers/ColorScheme.swift
import SwiftUI

/// SwiftUI `.colorScheme(_:)`. Forces the subtree to render in the given
/// `ColorScheme`, overriding the inherited system appearance.
enum RNWColorSchemeModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("colorScheme") { view, params, _ in
            guard let scheme = parse(params.string("value")) else { return view }
            return AnyView(view.colorScheme(scheme))
        }
    }

    static func parse(_ s: String?) -> ColorScheme? {
        switch s {
        case "light": return .light
        case "dark":  return .dark
        default:      return nil
        }
    }
}

File: src/modifiers/colorScheme.ts
import { createModifier } from './createModifier';

/// SwiftUI `ColorScheme` values.
export type ColorScheme = 'light' | 'dark';

/// SwiftUI `.colorScheme(_:)`. Forces this view's subtree to render with the
/// given color scheme, overriding the system appearance.
export function colorScheme(value: ColorScheme) {
  return createModifier('colorScheme', { value });
}
